src/System.Drawing.Common/src/System/Drawing/Pen.cs

```diff
@@ -27,6 +27,9 @@ public sealed partial class Pen : MarshalByRefObject, ICloneable, IDisposable
         private Color _color;
         private bool _immutable;
 
+        // Tracks whether the dash style has been changed to something else than Solid during the lifetime of this object.
+        private bool dashStyleWasChanged;
```

Comment:
Prefix this with an underscore. Also, could you rename this to something like "_dashStyleIsNotSolid" to better reflect the name.

An aside: I'm okay with doing this as an attempt to match behavior better, but it's not perfectly robust. It is legal to change the native properties of the Pen outside of the Pen class (through a PInvoke or otherwise), and this cached state will become stale. This is a common problem in the library, and not specific to this at all. I just wanted to point it out.